OpenRA.Mods.Common/Traits/World/SmudgeLayer.cs

```diff
@@ -150,8 +154,14 @@ public void AddSmudge(CPos loc)
 				return;
 
 			if (hasSmoke && Game.CosmeticRandom.Next(0, 100) <= Info.SmokeChance)
+			{
+				var maxOffsetDistance = Info.MaxSmokeOffsetDistance.Length;
+				var x = world.SharedRandom.Next(-maxOffsetDistance, maxOffsetDistance);
```

Comment:
⚠️ Desync alert!

We're using `Game.CosmeticRandom` to decide whether to show the smoke, so some clients will call this and others don't. We need to be using `SharedRandom` or `CosmeticRandom` consistently here. It probably makes sense to use `SharedRandom`?

Reply:
On second look, there are several other uses of `CosmeticRandom`/`LocalRandom` before we get to this point, so safer to avoid `SharedRandom` and keep it client-side.